### PR TITLE
Relay: add relay output invert function

### PR DESCRIPTION
**Pull request by LaneaLucy**
added a option for every relay to invert the output signal.
If enabled, Relay on would be Pin output off and Relay off would be Pin output on.

### CODE REVIEW

File: libraries/AP_Relay/AP_Relay.cpp [hunk 1]
```diff
@@ -408,6 +408,10 @@ void AP_Relay::set_pin_by_instance(uint8_t instance, bool value)
 
     const bool initial_value = get_pin(pin);
 
+    if (_params[instance].invert) {
```
Comment (IamPete1) [inline libraries/AP_Relay/AP_Relay.cpp:411]:
```suggestion
    if (_params[instance].invert > 0) {
```
We quite commonly also allow -1 for disabled although its not documented.

File: libraries/AP_Relay/AP_Relay.cpp [hunk 2]
```diff
@@ -496,6 +500,10 @@ bool AP_Relay::get(uint8_t instance) const
         return false;
     }
 
+    if (_params[instance].invert) {
```
Comment (IamPete1) [inline libraries/AP_Relay/AP_Relay.cpp:503]:
```suggestion
    if (_params[instance].invert > 0) {
```

**Review by IamPete1 — COMMENTED** [review]
Can also remove this comment:

https://github.com/ArduPilot/ardupilot/blob/f0fc447aed925cebe283344b1bc670aa7803606e/libraries/AP_Relay/AP_Relay.cpp#L366

It might also be worth adding a sentence to the `DEFAULT` parameter description to note that the default is also inverted if invert is set. Or I guess we could make the default ignore inversion.

**Comment by LaneaLucy** [comment]
> It might also be worth adding a sentence to the `DEFAULT` parameter description to note that the default is also inverted if invert is set.

Isn't it self understanding that invert option inverts the pin output, not the relay state and default option is the relay default state?

**Comment by rmackay9** [comment]
Thanks for the contribution!

Could you change the commits to start with "Relay:"?  We always try to start commit titles with the subsystem prefixed because it makes it instantly obvious from looking at a commit what part of the system it affects.  This makes backporting much easier.  .. and just to be clear, I mean the commit title, not the PR title.

**Comment by LaneaLucy** [comment]
> Could you change the commits to start with "Relay:"? 

i have no idea how to do it. if i try to do it in gitkraken, it makes a new commit in master...

**Comment by tridge** [comment]
i've squashed it down to one commit and force pushed

**Comment by tridge** [comment]
@rmackay9 would you prefer "INVERT" or "INVERTED" to "REVERSED" ?

**Comment by rmackay9** [comment]
@tridge,

I'd actually prefer we make it REVERSED to be consistent with RC_Channel and SRV_Channel libraries.  If we could wind back time, I'd probably make RC_Channel and SRV_Channels use "REVERSE" instread of "REVERSED" but that ship has sailed so let's go with REVERSED for this PR.

**Comment by rmackay9** [comment]
@LaneaLucy,

Great to hear that AP rover could use the L298N motors without a Lua script.  I'd really like to get a wiring diagram and setup instructions into our wiki.  I see we have some instructions linked from our wiki here https://ardupilot.org/rover/docs/common-brushed-motors.html but maybe we could replace those with a new page.  I could help put that page together I think but I'll need some help from you to confirm it's correct.

**Comment by LaneaLucy** [comment]
@rmackay9 should we discuss this on discord or should I write down it here?

**Comment by rmackay9** [comment]
@LaneaLucy,

Either place is fine with me.  [Perhaps discord is a little better](https://discuss.ardupilot.org/c/ardurover/rover-45/175) because other users may add their comments there as well.

**Comment by rmackay9** [comment]
Hi @LaneaLucy,

We discussed on the dev call and decided to go with _INVERTED.  Sorry for creating the extra work for you.

**Comment by tridge** [comment]
discussed on the dev call, decided INVERTED is better, I will rework the PR and force push for INVERTED

**Comment by tridge** [comment]
updated and marked for merge

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by LaneaLucy** [comment]
Nice

**Pull request merged**